### ISSUE TITLE
Problems with TRule Typescript definition and rendering keyframes

### ISSUE THREAD

**Issue by UberMouse** [rofrischmann/fela#625]
**Type**: Bug

| Packages | Version |
| -------- | ------- |
| fela     | 6.2.2   |

-------------------
#### Description
The Typescript typings for a Rule function are missing the optional argument for the renderer. We use this to render our keyframes into the functions style output. Maybe this isn't the correct method? The docs aren't clear.

Assuming it is a supported method, the typings for [TRule](https://github.com/rofrischmann/fela/blob/master/packages/fela/index.d.ts#L6) should be updated to include the renderer argument so having the argument doesn't cause a type error.
#### Expected Behavior
```javascript
import { IStyle, IRenderer } from 'fela';
import { createComponent } from 'react-fela';

const StyledFoo = (_props: SomeProps, renderer: IRenderer): IStyle => ({
  animation: `${renderer.renderKeyframe(animateIn, props)} 0.2s ease-in-out`,
});

// No error
export const Foo = createComponent(StyledFoo, 'div');
```

#### Actual Behavior
```javascript
import { IStyle, IRenderer } from 'fela';
import { createComponent } from 'react-fela';

const StyledFoo = (_props: SomeProps, renderer: IRenderer): IStyle => ({
  animation: `${renderer.renderKeyframe(animateIn, props)} 0.2s ease-in-out`,
});

// Error on StyledFoo
// TS2345: Argument of type '(props: Props & ThemeType, renderer: IRenderer) => IStyle' is not assignable to parameter of type 'Style<Props & ThemeType>'. 
//  Type '(props: Props & ThemeType, renderer: IRenderer) => IStyle' is not assignable to type 'StyleFunction<Props & ThemeType>'.
export const Foo = createComponent(StyledFoo, 'div');
```

**Comment by UberMouse** [comment]
I'm also wondering if the typing for `TKeyFrame` might be better served being
`export type TKeyFrame = {[keyFrame: string]: TRule`. The keys can only be keyframes, not styles right?

**Comment by rofrischmann** [comment]
Feel free to send a PR improving the TS definitions! I have no experience with TS at all.

**Comment by rofrischmann** [comment]
Fixed with #701
